library/core/class.request.php

```diff
@@ -100,16 +100,37 @@ public function __construct() {
      *
      * @param string? $assetRoot An asset root to set.
      * @return string Returns the current asset root.
+     *
+     * @deprecated 2.8. Use the explicit asset functions instead.
      */
     public function assetRoot($assetRoot = null) {
         if ($assetRoot !== null) {
-            $result = $this->_parsedRequestElement('AssetRoot', rtrim('/'.trim($assetRoot, '/'), '/'));
+            deprecated(__FUNCTION__, "setAssetRoot");
+            $this->setAssetRoot($assetRoot);
+            return $assetRoot;
         } else {
-            $result = $this->_parsedRequestElement('AssetRoot');
+            deprecated(__FUNCTION__, "addAssetRoot");
+            $result = $this->getAssetRoot();
         }
         return $result;
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function getAssetRoot() {
+        return $this->_parsedRequestElement('AssetRoot');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setAssetRoot(string $assetRoot) {
+        $this->_parsedRequestElement('AssetRoot', rtrim('/'.trim($assetRoot, '/'), '/'));
```

Comment:
I know you're copying forward existing code, but let's give it a little more breathing room.

```suggestion
        $this->_parsedRequestElement('AssetRoot', rtrim('/' . trim($assetRoot, '/'), '/'));
```

Reply:
I'm possibly overlooking something (and maybe this was discussed before), but the trimming seems a little redundant to me.

1. Provide an asset root (/path/to/assets/).
1. Trim off forward slashes from either end (path/to/assets).
1. Prepend a forward slash (/path/to/assets).
1. Trim any trailing forward slashes. But these should've been removed in step two, right?

Reply:
There's definitely a difference in the output https://repl.it/repls/OrchidThistleTerabyte.

Even if "not" having the extra trim is easier to follow it does adjust the way the paths are put together a good bit, and this type of thing affects all of the legacy asset system.

Changing that in such a big pull request like this would worry me a lot more than the configuration signature change.

Reply:
To recap: it looks like this is to force never ending a path with a slash. I'm assuming this was to make subsequent path concatenation easier. If the result of your path calculation ends up being "/", Vanilla normalizes that to "no path" (an empty string). Otherwise, you could end up with a path that looks like "//path/to/asset", where the first "/" is the "asset root".